src/store/addedSafesSlice.ts

```diff
@@ -110,15 +113,41 @@ export const selectAddedSafes = createSelector(
 export const addedSafesMiddleware: Middleware<{}, RootState> = (store) => (next) => (action) => {
   const result = next(action)
 
-  if (action.type === balancesSlice.actions.set.type) {
-    const state = store.getState()
-    const { data } = selectSafeInfo(state)
+  const state = store.getState()
 
-    const chainId = data?.chainId
-    const address = data?.address.value
+  switch (action.type) {
+    case addedSafesSlice.actions.addOrUpdateSafe.type: {
+      const { chainId, address } = action.payload.safe
 
-    if (chainId && address && action.payload.data) {
-      store.dispatch(updateAddedSafeBalance({ chainId, address, balances: action.payload.data }))
+      const addedSafes = selectAllAddedSafes(state)
+      const addedSafeAddresses = Object.keys(addedSafes?.[chainId] || {})
+
+      if (isAddedSafe(addedSafes, chainId, address.value) || addedSafeAddresses.length === 0) {
+        return
+      }
+
+      const event = OVERVIEW_EVENTS.ADDED_SAFES_ON_NETWORK
+
+      const currentChain = selectChainById(state, chainId)
+      const { chainName } = currentChain || {}
+
+      trackEvent({
+        ...event,
+        action: `${event.action} ${chainName}`,
+        label: addedSafeAddresses.length,
+      })
```

Comment:
I assume this part is tracking. Please encapsulate in a function to not mix it with other code in this middleware.

Reply:
This new `case` is tracking logic [you requested moved](https://github.com/safe-global/web-core/pull/371#discussion_r948729979). Do you want me to just extract the logic you highlighted or the entire `case` logic?

Reply:
Ah, the entire case. Then it's fine. Maybe just add a comment on top. This entire switch is quite big.